crates/rolldown_common/src/module/external_module.rs

```diff
@@ -11,25 +15,63 @@ pub struct ExternalModule {
   /// - Used for iife format to inject symbol and deconflict.
   /// - Used for for rewrite `import { foo } from 'external';console.log(foo)` to `var external = require('external'); console.log(external.foo)` in cjs format.
   pub namespace_ref: SymbolRef,
+  pub id: ArcStr,
   pub name: ArcStr,
+  pub identifier_name: ArcStr,
   pub import_records: IndexVec<ImportRecordIdx, ResolvedImportRecord>,
   pub side_effects: DeterminedSideEffects,
+  pub need_renormalize_render_path: bool,
 }
 
 impl ExternalModule {
   pub fn new(
     idx: ModuleIdx,
+    id: ArcStr,
     name: ArcStr,
+    identifier_name: ArcStr,
     side_effects: DeterminedSideEffects,
     namespace_ref: SymbolRef,
+    need_renormalize_render_path: bool,
   ) -> Self {
     Self {
       idx,
+      id,
       exec_order: u32::MAX,
       namespace_ref,
       name,
+      identifier_name,
       import_records: IndexVec::default(),
       side_effects,
+      need_renormalize_render_path,
+    }
+  }
+
+  pub fn get_import_path(&self, chunk: &Chunk) -> ArcStr {
+    if !self.need_renormalize_render_path {
+      return self.name.clone();
+    }
+    let mut target = self.name.as_str();
+    let mut importer = chunk
+      .preliminary_filename
+      .as_deref()
+      .expect("importer chunk should have preliminary_filename")
+      .to_string();
+    while target.starts_with("../") {
```

Comment:
This does not work on windows

Reply:
The `target` come form the `name`, it could be an absolute path or a normalized relative path. Here only handle the already normalized relative path.